### ISSUE TITLE
Mapping `0` to `^` screws up using a count containing a zero

### ISSUE THREAD

**Issue by J-Fields** [VSCodeVim/Vim#4184]
It's parsing `50G`, which should still go to the 50th line as a count of 5 then `0`, which is not how vim behaves and makes this sort of remap pretty much useless.

**Comment by michaelmoussa** [comment]
@J-Fields I just noticed I was having the same problem, found your open issue, then found [this one](https://github.com/VSCodeVim/Vim/issues/3349) later in my search.

Removing the `vim.normalModeKeyBindingsNonRecursive` block in my `settings.json` file fixed the issue for me, so maybe that will help for you.

Apparently I added this a _long_ time ago to allow me to use `j/k/0/$` movement in soft-wrapped lines and must not have needed to jump to a line number containing a `0` since then! Of course, this means I can't move within soft-wrapped lines using `j/k/0/$` anymore. However, that can be worked around by preceding the `j/k/0/$` commands with a `g`, while "I can't jump to any lines containing a `0`" problem can't as far as I know.
